server/plugins/api/index.ts

```diff
@@ -0,0 +1,12 @@
+import { GenericNotifier } from './GenericNotifier';
+import Facebook from "./Facebook";
+import LiveSite from "./LiveSite";
+import Slack from "./Slack";
+import Twitter from "./Twitter";
+
+export const mediaAPI: { [medium: string]: GenericNotifier } = {
```

Comment:
This is hiding type restrictions, for instance say `Facebook` had a:
```typescript
interface Config extends GenericConfig {
    message: string;
    post_type: string;
}
```

You could _still_ run `mediaAPI.FACEBOOK.sendMessage({ message: "" })`, no `post_type`!
We gotta think of a way to handle these configs better. For now it might be safer to use `config: any` and check types at runtime.

Reply:
I can't figure out how I would do this in a slick way with Typescript, but does the config check you mentioned below a workaround to this requirement? Either way, I'd like to see a good typescript solution to this problem that doesn't use the workaround. For now, idk either thing, however.